atom/browser/api/atom_api_web_contents_impl.cc

```diff
@@ -21,8 +24,15 @@ namespace api {
 
 void WebContents::DetachFromOuterFrame() {
   // See detach_webview_frame.patch on how to detach.
-  auto* impl = static_cast<content::WebContentsImpl*>(web_contents());
-  impl->GetRenderManagerForTesting()->RemoveOuterDelegateFrame();
+  DCHECK(content::GuestMode::IsCrossProcessFrameGuest(web_contents()));
+  int frame_tree_node_id =
+      static_cast<content::WebContentsImpl*>(web_contents())
+          ->GetOuterDelegateFrameTreeNodeId();
+  if (frame_tree_node_id != content::FrameTreeNode::kFrameTreeNodeInvalidId) {
```

Comment:
in what situation would `frame_tree_node_id` be the invalid ID? should we `DCHECK` or `CHECK` it?

Reply:
This specifically tries to obtain the node id of an outer webcontents, so it would return an invalid id when there is no outer webcontents associated for this content.